spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ClientStore.java

```diff
@@ -139,7 +161,7 @@ public String watchedKeyNames(ConfigStore store, Map<String, List<String>> store
         String watchedKey = store.getWatchedKey().trim();
         List<String> contexts = storeContextsMap.get(store.getEndpoint());
 
-        String watchedKeys = contexts.stream().map(ctx -> genKey(ctx, watchedKey))
+        String watchedKeys = contexts.stream().map(ctx -> genKey(ctx, watchedKey)).distinct()
                 .collect(Collectors.joining(","));
 
         if (watchedKeys.contains(",") && watchedKeys.contains("*")) {
```

Comment:
If seems this logic could be simplified. If I understood right, the logic here is: 
```
If we have more than one(not equal) or `!StringUtils.isNoneEmpty(watchedKey)`
     return "*"
else 
    return context+watchedKey
```

What do you think about this code to replace the logic from line 164 to 180?
```java
    private static final String ALL_KEYS = "*";
    ...
    private String supportedWatchedKey(List<String> contexts, String watchedKey) {
        if (watchedKey.contains(ALL_KEYS)){
            return ALL_KEYS;
        }

        List<String> contextsUniques = contexts.stream().distinct().collect(toList());
        if (contextsUniques.size() > 1) {
            LOGGER.warn("Multi keys not supported. It will use watch all keys(*) instead {}", Arrays.toString(contextsUniques.toArray()));
            return ALL_KEYS;
        }

        return contextsUniques.stream().findFirst().orElse("") + watchedKey;
    }
```